src/internal/consistenthashing/etcd.go

```diff
@@ -208,6 +205,27 @@ func (ring *Ring) getIndex(key string) int {
 	return index
 }
 
+func (ring *Ring) releaseLock(key string) error {
+	lockInfo, exists := ring.node.locks[key]
+	if !exists {
+		return nil
+	}
+	delete(ring.node.locks, key)
+	// The lock is released when either:
+	// - The lock context is canceled.
+	// - The unlock call completes successfully.
+	ctx := lockInfo.ctx
+	return backoff.RetryNotify(func() error {
+		if errors.Is(ctx.Err(), context.Canceled) {
+			return nil
+		}
+		return lockInfo.lock.Unlock(ctx)
+	}, backoff.NewInfiniteBackOff(), func(err error, d time.Duration) error {
```

Comment:
is there a concern with having an infinite backoff here? If we're already in the case where we're failing to unlock, shouldn't we be retrying with more urgency lol

Reply:
We are using an exponential backoff because that is our standard strategy for transient errors and we use the infinite backoff because we have to ensure that the lock is released.